src/node.rs

```diff
@@ -89,12 +94,21 @@ impl PbftNode {
             PbftMessageType::PrePrepare => {
                 let pbft_message: PbftMessage = extract_message(&msg)?;
 
+                if !verify_message_sender(&&pbft_message, sender_id) {
```

Comment:
Wildly out of scope for this PR, but I opened https://bitwiseio.atlassian.net/browse/IP-72 to track refactoring protobuf messages so that we don't have to have this check in each branch of the match statement, we could just have `extract_message` and `verify_message_sender` called once at the start of `on_peer_message`.

Reply:
Good to know! It is a bit cumbersome to have to make the same call in so many places, so that will be a nice improvement.